### PR TITLE
Implement per-workspace autotiling

### LINKED ISSUE #3: Add --version flag
Without using any extra dependencies as it is just one simple flag.

### PR DESCRIPTION
**Pull request by MaxVerevkin**
Closes #3 and #4.

This does introduce a new dependency: `clap`, but I think it's better that reinventing the wheel.

Also add `target` directory in `.gitignore`.

### CODE REVIEW

**Comment by ammgws** [comment]
That looks about right to me.

Cheers!

File: src/main.rs [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+fn switch_splitting(conn: &mut Connection, workspaces: &[String]) -> Result<(), String> {
+    // Check if focused workspace is in "allowed list".
+    // If `workspaces` is empty, skip allow all workspaces.
+    if !workspaces.is_empty() {
+        for workspace in conn
+            .get_workspaces()
+            .map_err(|_| "get_workspaces() failed")?
+        {
+            if workspace.focused {
+                if workspaces.contains(&workspace.name) {
```
Comment (ammgws) [inline src/main.rs:15]: Should we check .num instead?

Reference: https://github.com/nwg-piotr/autotiling/pull/22#pullrequestreview-563904491

File: src/main.rs [hunk 2]
```diff
@@ -40,28 +59,44 @@ fn switch_splitting(conn: &mut Connection) -> Result<(), String> {
...
+                if let WindowChange::Focus = e.change {
                     //we can not use the e.container because the data is stale
                     //if we compare that node data with the node given from get_tree() after we
                     //delete a node we find that the e.container.rect.height and e.container.rect.width are stale
                     //and therefore we make the wrong decision on which layout our next window
                     //should be
-                    if let Err(err) = switch_splitting(&mut conn) {
-                        println!("err: {}", err);
+                    if let Err(err) = switch_splitting(&mut conn, &workspaces) {
+                        eprintln!("err: {}", err);
```
Comment (ammgws) [inline src/main.rs:93]: Good catch

File: src/main.rs (hunk 1 above)
Comment (MaxVerevkin) [reply src/main.rs:15]: It turns out I've been using this feature the wrong way. I'll open new PR.

**Pull request merged**